tests/decorators/tests.py

```diff
@@ -474,6 +474,25 @@ def method(self):
         self.assertIsNotNone(func_module)
 
 
+class SyncAndAsyncDecoratorTests(TestCase):
+    """
+    Tests to make sure all builtin decorators declare themselves as sync and
+    async capable.
+    """
+
+    def test_decorators_sync_and_async_capable(self):
+        decorators = (
+            cache_page,
+            cache_control,
+            never_cache,
+        )
+
+        for decorator in decorators:
+            with self.subTest(decorator):
+                self.assertTrue(decorator.sync_capable)
+                self.assertTrue(decorator.async_capable)
```

Comment:
This isn't the relevant test. 

A decorator is applied to a view, which is passed through `iscoroutinefunction` in `BaseHandler._get_response` and `_get_response_async`. (This is **inside** the middleware chain, the loading of which is what uses the `sync_capable` and `async_capable` markers.)

The relevant check is like: 

```python
decorated = decorator(async_view)
assert iscoroutinefunction(decorated)
``` 

... and vice versa for a sync view, for each decorator.